src/__tests__/verifyKeyMiddleware.ts

```diff
@@ -0,0 +1,131 @@
+import express, { Request, Response } from 'express';
+import { InteractionResponseType, InteractionType, verifyKeyMiddleware } from '../index';
+import { AddressInfo } from 'net';
+import {
+  applicationCommandRequestBody, generatedInvalidKeyPair,
+  generatedValidKeyPair,
+  pingRequestBody,
+  sendExampleRequest,
+  signRequestWithKeyPair,
+} from './utils/SharedTestUtils';
+import * as http from 'http';
+import DoneCallback = jest.DoneCallback;
+
+const expressApp = express();
+
+const exampleApplicationCommandResponse = {
+  type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
+  data: {
+    content: 'Hello world'
+  }
+};
+
+expressApp.post('/interactions', verifyKeyMiddleware(Buffer.from(generatedValidKeyPair.publicKey).toString('hex')), (req: Request, res: Response) => {
+  const interaction = req.body;
+  if (interaction.type === InteractionType.COMMAND) {
+    res.send(exampleApplicationCommandResponse);
+  }
+});
+
+let expressAppServer: http.Server;
+let exampleInteractionsUrl: string;
+
+beforeAll(async (done: DoneCallback) => {
+  await new Promise<void>((resolve) => {
+    expressAppServer = expressApp.listen(0);
+    resolve();
+  });
+  exampleInteractionsUrl = 'http://localhost:' + (expressAppServer.address() as AddressInfo).port + '/interactions';
+  done();
+});
+
+describe('verify key middleware', () => {
+
+  it('valid', async (done: DoneCallback) => {
+    // Sign and verify a valid request
+    const signedRequest = signRequestWithKeyPair(pingRequestBody, generatedValidKeyPair.secretKey);
+    const exampleRequestResponse = await sendExampleRequest(exampleInteractionsUrl, {
+      'x-signature-ed25519': signedRequest.signature,
+      'x-signature-timestamp': signedRequest.timestamp,
+      'content-type': 'application/json'
+    }, signedRequest.body)
+    expect(exampleRequestResponse.status).toBe(200);
+    done();
```

Comment:
Isn't `done()` unnecessary in an async test?  https://jestjs.io/docs/en/asynchronous#asyncawait

Reply:
The reason for this was that it is using callbacks. I don't exactly remember why I chose this method but if you would like, I could convert it away from using callbacks. https://jestjs.io/docs/en/asynchronous#callbacks